Core/src/IfcGeometryConverter/SplineConverter.h

```diff
@@ -147,12 +147,149 @@ namespace OpenInfraPlatform
 					 * \internal	The Code of the function is in the commented out part at the end of the file.
 					 */
 					void convertIfcBSplineSurface(
-						const std::shared_ptr<typename IfcEntityTypesT::IfcBoundedSurface>& splineSurface,
-						const std::vector<std::vector<carve::geom::vector<3>>>& controlPoints,
+						const EXPRESSReference<typename IfcEntityTypesT::IfcBSplineSurface>& splineSurface,
+						const carve::math::Matrix& pos, // AT THE MOMENT, NO IDEA WHAT THIS IS
+						//const std::vector<std::vector<carve::geom::vector<3>>>& controlPoints,
 						std::shared_ptr<carve::input::PolylineSetData>& polylineData) const throw(...)
 					{
-						// not implemented, code in commented out part at the end of the file
-						throw oip::UnhandledException(splineSurface);
+
+						if (splineSurface.isOfType<typename IfcEntityTypesT::IfcBSplineSurfaceWithKnots>())
+						{
+							const EXPRESSReference<typename IfcEntityTypesT::IfcBSplineSurfaceWithKnots>& bsplineSurfaceWithKnots =
+								splineSurface.as<typename IfcEntityTypesT::IfcBSplineSurfaceWithKnots>();
+
+							// obtain degree of both b-spline curves
+							const int degreeU = bsplineSurfaceWithKnots->UDegree;
+							const int orderU = degreeU + 1;
+							const int degreeV = bsplineSurfaceWithKnots->VDegree;
+							const int orderV = degreeV + 1;
+							// obtain number of control points
+							//const int numControlPointsU = controlPoints.size();
+							//const int numControlPointsV = controlPoints[0].size();
+
+							// load array of control points from ifc-entity
+							// simplified, it's in the type EXPRESSContainer<EXPRESSContainer<IfcCartesianPoint>>
+							const auto controlPointList = bsplineSurfaceWithKnots->ControlPointsList;
+							// temporary control point container of type std::vector<std::vector<..>> which is needed for converter function
+							std::vector<std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcCartesianPoint>>> controlPointListVector;

```

Comment:
An `EXPRESSContainer` is a `std::vector` so there should be no need to transform here. (I may be wrong, though).

Reply:
By puting `bsplineSurfaceWithKnots->ControlPointsList` directly into a `std::vector<std::vector<...>>`, the compiler complained " 'initializing': cannot convert from 'const OpenInfraPlatform::EarlyBinding::EXPRESSContainer<T,0,18446744073709551615>' to 'std::vector<std::vector<ValueType,std::allocator<_Ty>>,std::allocator<std::vector<_Ty,std::allocator<_Ty>>>>' "

However, at least this specific case is resolved by the conversation next below (about changing the function parameter type).